cpp/mcap/include/mcap/mcap.hpp

```diff
@@ -1,12 +1,22 @@
 #pragma once
 
+#include "errors.hpp"
+#include <climits>
+#include <iostream>
+#include <optional>
 #include <string>
+#include <string_view>
 #include <unordered_map>
+#include <unordered_set>
 #include <vector>
 
 namespace mcap {
 
-constexpr char SpecVersionChar = '0';
+#define LIBRARY_VERSION "0.0.1"
+
+constexpr char SpecVersion = '0';
+constexpr char LibraryVersion[] = LIBRARY_VERSION;
+constexpr char Magic[] = {char(137), 77, 67, 65, 80, SpecVersion, 13, 10};  // "\x89MCAP0\r\n"
```

Comment:
why does 137 need a cast? does something happen without the cast? maybe this should all be `unsigned char[]`?

Reply:
`char` is a signed 8-bit value, so 137 (being larger than 127) overflows and clang will warn about this. Sadly, our Dockerfile-based build seems to swallow up warnings but I see the warning when I compile directly on MacOS using:

```
Homebrew clang version 13.0.0
Target: arm64-apple-darwin21.1.0
Thread model: posix
InstalledDir: /opt/homebrew/opt/llvm/bin
```

I could make it an unsigned char array, but then I need to typecast when I do an ostream `write()` which wants a char array.

Reply:
> our Dockerfile-based build seems to swallow up warnings

😱